### DISCUSSION TITLE
Occasionally getting npm ERR! code ERR_SOCKET_TIMEOUT

### DISCUSSION THREAD

**Discussion by nilkemistry** [community/community#51258] [Actions]
### Select Topic Area

Bug

### Body

Occasionally we are facing this issue in executing the build jobs.

![image](https://user-images.githubusercontent.com/30195517/228326492-73be6f04-587b-4538-ba05-be497fe21465.png)

Could you please suggest, why we are getting this issue once in a while and how can we avoid it in future?

**Comment by vrvrkrishnan** [comment]
We experience this issue during the daytime peak business hours in the afternoon & evening.

**Comment by nilkemistry** [comment]
- We have set build job timeout for 15 minutes in workflow script file but still it is failing before that time. 

name: Deploy
    runs-on: ubuntu-latest #To change, self hosted Github runners
    timeout-minutes: 15
    environment: development

- We have also provided option for retry in the Workflow file but still it is not retrying and failing once in while
   - run: npm config set fetch-retries 5
      - run: npm config set fetch-retry-factor 20
      - run: npm config set fetch-retry-maxtimeout 120000
      - run: npm config set fetch-retry-mintimeout 20000
      - run: npm config set fetch-timeout 600000

Could anyone please help on this?
@vrvrkrishnan
